packages/nocodb/src/lib/version-upgrader/ncFilterUpgrader_0104004.ts

```diff
@@ -0,0 +1,380 @@
+          filter.id,
+          {
+            ...filter,
+            comparison_op: 'notchecked',
+            value: '',
+          },
+          ncMeta
+        )
+      );
+    } else {
+      // invalid value - good to delete
+      actions.push(await Filter.delete(filter.id, ncMeta));
+    }
+  }
+  return actions;
+};
+
+async function migrateFilters(ncMeta: NcMetaIO) {
+  const filters = await ncMeta.metaList2(null, null, MetaTable.FILTER_EXP);
+  let actions = [];
+  for (const filter of filters) {
+    const col = await Column.get({ colId: filter.fk_column_id }, ncMeta);
+    if (!col || !col.uidt) {
+      // column not found -> skip
+      continue;
+    }
+    if (
+      [
+        UITypes.SingleLineText,
+        UITypes.LongText,
+        UITypes.PhoneNumber,
+        UITypes.Email,
+        UITypes.URL,
+      ].includes(col.uidt)
+    ) {
+      actions = await removeArithmeticFilters(filter, actions, ncMeta);
+    } else if (
+      [
+        // numeric fields
+        UITypes.Duration,
+        UITypes.Currency,
+        UITypes.Percent,
+        UITypes.Number,
+        UITypes.Decimal,
+        UITypes.Rating,
+        UITypes.Rollup,
+        // select fields
+        UITypes.SingleSelect,
+      ].includes(col.uidt)
+    ) {
+      actions = await removeLikeFilters(filter, actions, ncMeta);
+      actions = await migrateNullAndEmptyToBlankFilters(
+        filter,
+        actions,
+        ncMeta
+      );
+    } else if (col.uidt === UITypes.Checkbox) {
+      actions = await migrateToCheckboxFilter(filter, actions, ncMeta);
+    } else if (col.uidt === UITypes.MultiSelect) {
+      actions = await removeLikeFilters(filter, actions, ncMeta);
+      actions = await migrateNullAndEmptyToBlankFilters(
+        filter,
+        actions,
+        ncMeta
+      );
+      actions = await migrateMultiSelectEq(filter, actions, col, ncMeta);
+    } else if (col.uidt === UITypes.Attachment) {
+      actions = await removeArithmeticFilters(filter, actions, ncMeta);
+      actions = await removeEqualFilters(filter, actions, ncMeta);
+      actions = await migrateNullAndEmptyToBlankFilters(
+        filter,
+        actions,
+        ncMeta
+      );
+    } else if (col.uidt === UITypes.LinkToAnotherRecord) {
+      actions = await removeArithmeticFilters(filter, actions, ncMeta);
+      actions = await migrateNullAndEmptyToBlankFilters(
+        filter,
+        actions,
+        ncMeta
+      );
+    } else if (col.uidt === UITypes.Lookup) {
+      actions = await removeArithmeticFilters(filter, actions, ncMeta);
+      actions = await migrateNullAndEmptyToBlankFilters(
+        filter,
+        actions,
+        ncMeta
+      );
+    } else if (col.uidt === UITypes.Duration) {
+      actions = await removeLikeFilters(filter, actions, ncMeta);
+      actions = await migrateNullAndEmptyToBlankFilters(
+        filter,
+        actions,
+        ncMeta
+      );
+    }
+  }
+  await Promise.all(actions);
```

Comment:
Here `actions` is not a collection of promises so `await Promise.all(actions);` doesn't make any sense. Everywhere when pushing to `actions` you are doing an `await` as well so it's already completed and the result is not a promise.


One sample : 
```
actions.push(
      await Filter.update(
        filter.id,
        {
          ...filter,
          comparison_op: 'anyof',
          value: newFilterValue,
        },
        ncMeta
      )
    );
```